rust/core-lib/src/view.rs

```diff
@@ -882,6 +883,19 @@ impl View {
             self.set_selection(text, occ);
         }
     }
+
+    pub fn get_line_range(&self, text: &Rope, region: &SelRegion) -> Range<usize> {
+        let (first_line, _) = self.offset_to_line_col(text, region.min());
+        let (last_line, last_col) =
+            self.offset_to_line_col(text, region.max());
+        let last_line = if last_col == 0 && last_line > first_line {
+            last_line - 1
+        } else {
+            last_line
+        };
+        let line_range = first_line..(last_line + 1);
```

Comment:
Another style point, can just return rather than binding to a variable, esp. since the name of the variable is pretty much the same as the name of the function.